### PR TITLE
LibCompress: Port `ZlibDecompressor` to `AK::Stream` (and other small changes)

### PR DESCRIPTION
**Pull request by timschumi**
These I had made when starting to try and untangle LibHTTP's decompressor setup, but that ultimately got nowhere. Still, these three commits are useful on their own.

### CODE REVIEW

File: Meta/Lagom/Fuzzers/FuzzZlibDecompression.cpp [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
  */
 
+#include <AK/MemoryStream.h>
 #include <LibCompress/Zlib.h>
 #include <stdio.h>
 
 extern "C" int LLVMFuzzerTestOneInput(uint8_t const* data, size_t size)
 {
-    (void)Compress::ZlibDecompressor::decompress_all(ReadonlyBytes { data, size });
+    auto stream = MUST(try_make<FixedMemoryStream>(ReadonlyBytes { data, size }));
```
Comment (LucasChollet) [inline Meta/Lagom/Fuzzers/FuzzZlibDecompression.cpp:13]: I guess you can use make directly now

File: Tests/LibCompress/TestZlib.cpp [hunk 2]
```diff
@@ -20,8 +21,10 @@ TEST_CASE(zlib_decompress_simple)
 
     const u8 uncompressed[] = "This is a simple text file :)";
 
-    auto const decompressed = Compress::ZlibDecompressor::decompress_all(compressed);
-    EXPECT(decompressed.value().bytes() == (ReadonlyBytes { uncompressed, sizeof(uncompressed) - 1 }));
+    auto stream = MUST(try_make<FixedMemoryStream>(compressed));
```
Comment (LucasChollet) [inline Tests/LibCompress/TestZlib.cpp:24]: same here

File: Tests/LibCompress/TestZlib.cpp [hunk 3]
```diff
@@ -73,7 +76,8 @@ TEST_CASE(zlib_decompress_with_missing_end_bits)
         0x44, 0x11, 0x10, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
     };
 
-    auto const maybe_decompressed = Compress::ZlibDecompressor::decompress_all(compressed);
-    EXPECT(maybe_decompressed.has_value());
-    EXPECT_EQ(maybe_decompressed.value().span(), decompressed.span());
+    auto stream = MUST(try_make<FixedMemoryStream>(compressed));
```
Comment (LucasChollet) [inline Tests/LibCompress/TestZlib.cpp:79]: and here :)

File: Userland/Libraries/LibGfx/Font/WOFF/Font.cpp [hunk 4]
```diff
@@ -120,12 +121,18 @@ ErrorOr<NonnullRefPtr<Font>> Font::try_load_from_externally_owned_memory(Readonl
         if (font_buffer_offset + orig_length > font_buffer.size())
             return Error::from_string_literal("Uncompressed WOFF table too big");
         if (comp_length < orig_length) {
-            auto decompressed = Compress::ZlibDecompressor::decompress_all(buffer.slice(offset, comp_length));
-            if (!decompressed.has_value())
-                return Error::from_string_literal("Could not decompress WOFF table");
-            if (orig_length != decompressed->size())
+            auto compressed_data_stream = TRY(try_make<FixedMemoryStream>(buffer.slice(offset, comp_length)));
+            auto decompressor_or_error = Compress::ZlibDecompressor::create(move(compressed_data_stream));
+            if (decompressor_or_error.is_error())
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/Font/WOFF/Font.cpp:126]: Why aren't you using a `TRY()`?

Comment (timschumi) [reply Userland/Libraries/LibGfx/Font/WOFF/Font.cpp:126]: Why indeed. I'm sure I thought that I had a reason for doing that.

File: Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp [hunk 5]
```diff
@@ -1443,12 +1459,15 @@ ErrorOr<Optional<ReadonlyBytes>> PNGImageDecoderPlugin::icc_data()
...
     if (m_context->embedded_icc_profile.has_value()) {
         if (!m_context->decompressed_icc_profile.has_value()) {
-            auto result = Compress::ZlibDecompressor::decompress_all(m_context->embedded_icc_profile->compressed_data);
-            if (!result.has_value()) {
-                m_context->embedded_icc_profile.clear();
-                return Error::from_string_literal("PNGImageDecoderPlugin: Decompression of ICC profile failed");
-            }
-            m_context->decompressed_icc_profile = move(*result);
+            auto compressed_data_stream = make<FixedMemoryStream>(m_context->embedded_icc_profile->compressed_data);
+            auto decompressor_or_error = Compress::ZlibDecompressor::create(move(compressed_data_stream));
```
Comment (LucasChollet) [inline Userland/Libraries/LibGfx/ImageFormats/PNGLoader.cpp:1463]: Pretty sure it's the same reason that forced me to not mention these on the first review.

**Review by LucasChollet — APPROVED** [review]

**Pull request merged**